### DISCUSSION TITLE
Quarto documentation example no longer maintained

### DISCUSSION THREAD

**Discussion by axiomtutor** [quarto-dev/quarto-cli#7920] [Q&A]
### Bug description

I'm trying to follow the documentation's example of installing RevealJS plugins here:  https://quarto.org/docs/presentations/revealjs/advanced.html#reveal-plugins

I'm trying to use this as a model for installing the "animated" plugin here:  https://github.com/rogeralmeida/revealjs-animated

So first thing is that I see on another page of the documentation (https://quarto.org/docs/extensions/) "Below is a listing of available extensions (please [let us know](https://github.com/quarto-dev/quarto-web/tree/main/docs/extensions/listings) if you have an extension you’d like to see added to the list)."  So on that note, if yall wanted to add this plugin I'd be happy to have it! :)

But in the mean time, I'm trying to follow the example.  However, the revealjs-animated files doesn't have an obvious main `.js` file, so I'm not sure (1) what the `script:` field in my `plugin.yml` file should point to, and (2) whether I need other parts of the YML for the other resources.  I'd guess either `dist\revealjs-animated.js` or `revealjs-animated.js` is right for the `script:` field, but after trying both it didn't work.  But I don't know if it didn't work because of this or some other guess I made along the way.

Here's my directory structure:

```
maindir 
|- _extensions
  |- (... stuff from other extensions)
|- animated
  |- (... all the stuff I downloaded and extracted for this extension, plus the `plugin.yml` file.)
|- _quarto
|- presentation.qmd
|- presentation
```

Here is my `plugin.yml`:

```
name: RevealJsAnimation
script: [dist\revealjs-animated.js]
```

and here is my `presentation.qmd`:

```
---
title: "title" 
author: "User Name" 
format:    
  revealjs:     
    chalkboard: true 
    revealjs-plugins:
     - dist\revealjs-animated.js
editor: visual 
---

# TEST


```

and here's the error report:

```
ERROR: Specified Reveal plugin directory 'dist\revealjs-animated.js' does not exist.

Stack trace:
    at pluginFromBundle (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:111904:15)
    at revealPluginExtras (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:111720:55)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async Object.formatExtras (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:112199:38)
    at async runPandoc (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:68549:60)
    at async renderPandoc (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:69843:26)
    at async Object.onRender (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:76974:36)
    at async renderFileInternal (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:76951:17)
    at async renderFiles (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:76775:17)
    at async renderProject (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:77102:25)
```

(please note I'm not a developer so I'm just doing my best to figure out how these things work, sorry!)

### Steps to reproduce

Hopefully downloading the extension, extracting, making the source files, and rendering, would reproduce this on anyone else's machine.

### Expected behavior

Working.

### Actual behavior

Not working.

### Your environment

Windows 11 on a Surface Book 3.

### Quarto check output

```
Quarto 1.4.523
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.9: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.523
      Path: C:\Users\username\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: v2023.12
      Chromium: 869685

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\username\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2023

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.4
      Path: C:/Program Files/Python311/python.exe
      Jupyter: 5.3.1
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.3.1
      Path: C:/PROGRA~1/R/R-43~1.1
      LibPaths:
        - C:/Users/username/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.1/library
      knitr: 1.44
      rmarkdown: 2.24

[>] Checking Knitr engine render......OK
```

**Comment by axiomtutor** [comment]
Oops, I just realized that the title for this post is not accurate -- I initially titled it when I thought that a link in the docs was dead or misdirected.  Only later did I hit the wall that I'm describing here but forgot to change the title at that point.

**Reply by axiomtutor** [reply]
Also, I don't know how big of an ask it is to add plugins to the standard Quarto library -- but if it's a small ask, then could we also add the Manim plugin?  https://www.manim.community/plugin/manim-revealjs/

**Reply by mcanouil** [reply]
The purpose of extensions is to allow user to add whatever they want without putting the burden of maintaining the extensions by the Quarto team which would mean less time to work on Quarto itself.

There are several examples of extensions:
- https://github.com/mcanouil/quarto-spotlight
- https://github.com/quarto-ext/pointer
- https://github.com/quarto-ext/attribution

You need to learn what Reveal.js plugins are, their structure and understand those you want to adapt them as Quarto extensions.
I'm afraid, it is very unlikely someone will create the extensions if not you, *i.e*., nobody spend time on something they don't use.

See the documentation on creating Quarto extension for Reveal.js plugins: <https://quarto.org/docs/extensions/revealjs.html>.

**Reply by axiomtutor** [reply]
My apologies @mcanouil if I misunderstood, but when I read 

> please [let us know](https://github.com/quarto-dev/quarto-web/tree/main/docs/extensions/listings) if you have an extension you’d like to see added to the list

here: https://quarto.org/docs/extensions/

I was unsure if my request to add this extension was the kind of thing that the page had in mind.

**Reply by mcanouil** [reply]
This means, if you create an extension, click on the link and add the GitHub repository of your Quarto extension to the listing (the appropriate YAML file) via a Pull Request.
It does not mean please request Quarto team to develop extensions.

> if you have an extension you’d like to see added to the list

Here, you don't have an extension.
You have merely an idea for an extension which as I said is unlikely to be developed by Quarto team unless there is a "high interest" by the community.
